Essentials/src/main/java/com/earth2me/essentials/commands/Commandhelpop.java

```diff
@@ -20,25 +23,37 @@ public Commandhelpop() {
     @Override
     public void run(final Server server, final User user, final String commandLabel, final String[] args) throws Exception {
         user.setDisplayNick();
-        final String message = sendMessage(server, user.getDisplayName(), args);
+        final String message = sendMessage(server, user, args);
         if (!user.isAuthorized("essentials.helpop.receive")) {
             user.sendMessage(message);
         }
     }
 
     @Override
     public void run(final Server server, final CommandSource sender, final String commandLabel, final String[] args) throws Exception {
-        sendMessage(server, Console.DISPLAY_NAME, args);
+        sendMessage(server, Console.getInstance(), args);
     }
 
-    private String sendMessage(final Server server, final String from, final String[] args) throws Exception {
+    private String sendMessage(final Server server, final IMessageRecipient from, final String[] args) throws Exception {
         if (args.length < 1) {
             throw new NotEnoughArgumentsException();
         }
-        final String message = tl("helpOp", from, FormatUtil.stripFormat(getFinalArg(args, 0)));
-        ess.getLogger().log(Level.INFO, message);
-        ess.broadcastMessage("essentials.helpop.receive", message);
-        return message;
+        final String message = FormatUtil.stripFormat(getFinalArg(args, 0));
+        final String finalMessage = tl("helpOp", from.getDisplayName(), message);
+        ess.getLogger().log(Level.INFO, finalMessage);
+        ess.broadcastMessage("essentials.helpop.receive", finalMessage);
+
+        final HelpopMessageSentEvent sentEvent = new HelpopMessageSentEvent(from, getOnlineHelpopRecipients(), message);
+        ess.getServer().getPluginManager().callEvent(sentEvent);
+
+        return finalMessage;
+    }
+
+    private List<? extends IMessageRecipient> getOnlineHelpopRecipients() {
+        return ess.getOnlinePlayers().stream()
+                .filter(player -> player.hasPermission("essentials.helpop.receive"))
+                .map(ess::getUser)
+                .collect(Collectors.toList());
```

Comment:
don't use streams and this list isn't mutable. changes to the list should effect who is sent the message.

Reply:
This event is currently called after the message has been sent, and I think you want to have an event before it is sent. What should I do?

Reply:
Correct, make it fire before. I guess you could rename the event to `HelpopMessageSendEvent` to better illustrate that. Unless there is a benefit to have the event send after that I don't see?

Reply:
If it's not cancellable I don't see any difference either. Will do it soon

Reply:
Still, how should I make a modifiable list of recipients properly?

Reply:
> Still, how should I make a modifiable list of recipients properly?

create an array list, do a for loop with all online players, add any player with the permission to the list, use the list in the event, call the event, then use the list to send the message out.

Reply:
Done!